test/test_files/function/hash/utility/coalesce.test

```diff
@@ -0,0 +1,172 @@
+-GROUP CoalesceFunctionTests
+-DATASET CSV empty
+
+--
+
+-CASE ConstantTests
+
+-LOG NoArgumentError
```

Comment:
I need to see a few more tests. On tinysnb, run `COALESCE` after a `MATCH (a:person)-[:knows]->(b:person)` and maybe also add some more query edges to it. This is to test different combination of parameter states.

Test coalesce of nested data types, LIST, STRUCT, MAP NODE, REL.

Test `RETURN ifNULL`